### DISCUSSION TITLE
Delayed job/workflow recommendation

### DISCUSSION THREAD

**Discussion by 0xmax** [supabase/supabase#8742] [Questions]
Hi, I did not find any previous discussion here. I would like to perform actions after database events. E.g send emails, create data or similar. What is the currently suggested and somewhat production ready recommendation for this? I had a look at webhooks but this is not ready for production yet and might be a bit brittle for critical tasks like seeding user data. Any pointers are much appreciated! The webapp is built with Next.js and deployed on Vercel. 

Thank you,

**Comment by saltcod** [comment]
Hey @0xmax !   Wonder would realtime work for this purpose? Subscribing to changes and reacting to them.

**Comment by GaryAustin1** [comment]
Besides webhooks there is just using a trigger function and the http extension.  
https://supabase.com/blog/postgres-as-a-cron-server

**Comment by 0xmax** [comment]
@GaryAustin1 Thank you, already using pg_cron but ideally I could have the workflow in source code. Also the minimum granularity of one minute is not ideal. 

@saltcod Thank you, I did not think of that. I think this should work. Stumbled upon this: https://github.com/supabase/workflows but it does not seem to be in development anymore.

**Reply by GaryAustin1** [reply]
I was more linking that as it showed using the http extension.  You use it in a trigger function on your table.  That was the quickest reference I could find.
